client/consensus/grandpa/delayKind.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+// DelayedKinds Kinds of delays for pending changes.
+type DelayedKinds interface {
+	Finalized | Best
+}
+
+// DelayKind struct to represent DelayedKinds
+type DelayKind struct {
+	value interface{}
+}
+
+func setDelayKind[T DelayedKinds](delayKind *DelayKind, val T) {
+	delayKind.value = val
+}
+
+func newDelayKind[T DelayedKinds](val T) DelayKind {
+	delayKind := DelayKind{}
+	setDelayKind(&delayKind, val)
+	return delayKind
+}
+
+// Finalized Depth in finalized chain.
+type Finalized struct{}
+
+// Best Depth in best chain. The median last finalized block is calculated at the time the
+// hashNumber was signaled.
+type Best struct {
+	medianLastFinalized uint
```

Comment:
Im actually unsure how to do this is this case. Since DelayedKinds is itself used as a generic type, we have something like

```
func setDelayKind[T DelayedKinds](delayKind *DelayKind, val T) {
	delayKind.value = val
}
```
and it doesn't seem to be valid to add generic constraint to the generic constraint, something like 


```
func setDelayKind[T DelayedKinds[N or N any, both don't work](delayKind *DelayKind, val T) {
	delayKind.value = val
}
```

Do you know how I can do this? Perhaps can also just use a VDT instead